src/adapter/src/catalog/builtin.rs

```diff
@@ -1389,6 +1389,14 @@ pub static MZ_KAFKA_CONNECTIONS: Lazy<BuiltinTable> = Lazy::new(|| BuiltinTable
         .with_column("sink_progress_topic", ScalarType::String.nullable(false)),
     is_retained_metrics_object: false,
 });
+pub static MZ_KAFKA_SOURCES: Lazy<BuiltinTable> = Lazy::new(|| BuiltinTable {
+    name: "mz_kafka_sources",
+    schema: MZ_CATALOG_SCHEMA,
+    desc: RelationDesc::empty()
+        .with_column("id", ScalarType::String.nullable(false))
+        .with_column("group_id_base", ScalarType::String.nullable(false)),
```

Comment:
Modest preference for the suggested `group_id_prefix` as well.

Though honestly I think the right strategy depends on how we implement replicated sources and sinks, and we haven't designed that yet! So if possible I suggest throwing this into `MZ_INTERNAL_SCHEMA` instead so we don't have to commit to getting this right forever immediately...

Reply:
Oh it's far more confusing than that! We accept a `GROUP ID PREFIX` option when you create the source. So I was thinking that `group_id_prefix` would represent the user-supplied group ID prefix, if any, and `group_id_base` would represent as much of the group ID as we could show without knowing the replica ID.

That's a lot to explain. Agree we should put this in `mz_internal`, now that you mention it...

Reply:
TIL, thanks!

Reply:
ill move it to `internal`, and leave it as `base`

Reply:
We're working through the explanation in the docs PR here!

<img width="876" alt="image" src="https://github.com/MaterializeInc/materialize/assets/882976/ccc11e54-babd-4dcd-a962-3229f2b42a81">